hack/e2e/test.sh

```diff
@@ -195,7 +194,40 @@ kubectl -n team-1 wait certificate app-0 --for=condition=Ready
 # Parse logs as JSON using jq to ensure logs are all JSON formatted.
 # Disable pipefail to prevent SIGPIPE (141) errors from tee
 # See https://unix.stackexchange.com/questions/274120/pipe-fail-141-when-piping-output-into-tee-why
+set +o pipefail
 kubectl logs deployments/venafi-kubernetes-agent \
         --follow \
         --namespace venafi \
     | timeout 60 jq 'if .msg | test("Data sent successfully") then . | halt_error(0) end'
+set -o pipefail
+
+# Create a unique TLS Secret and wait for it to appear in the Venafi certificate
+# inventory API. The case conversion is due to macOS' version of uuidgen which
+# prints UUIDs in upper case, but DNS labels need lower case characters.
+commonname="venafi-kubernetes-agent-e2e.$(uuidgen | tr '[:upper:]' '[:lower:]').example.com"
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/tls.key -out /tmp/tls.crt -subj "/CN=$commonname" 2>/dev/null
```

Comment:
@maelvls The additional suffix `example.com` caused an openssl error on my laptop, so I'm going to remove it.

```
$ openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/tls.key -out /tmp/tls.crt -subj /CN=venafi-kubernetes-agent-e2e.2294a4d1-275d-415d-a741-5ec4b514636e.example.com
...
40C7087C547F0000:error:06800097:asn1 encoding routines:ASN1_mbstring_ncopy:string too long:../crypto/asn1/a_mbstr.c:106:maxsize=64
req: Error adding subject name attribute "/CN=venafi-kubernetes-agent-e2e.2294a4d1-275d-415d-a741-5ec4b514636e.example.com"
```

Furthermore, the `2>/dev/null` redirect was obscuring that error, so I've removed the redirection.

FYI I originally added the redirect, to hide the distracting output of openssl:

```console
.....+......+.+..+...+...+.+...+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*..+........+...+.+......+.........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.....+...+..................+.............+.....+.+........+......+.........+.........................+......+..+.........+.+...+.........+......+.........+......+......+...+..+.+..+.......+........+.+..............+..........+.................+......+....+........+..........+.....+...+...+......+.........+......+....+..+..........+........+....+..+.+...............+.........+.........+..+....+......+...+.....+.+.....+...+.+............+..+...+...+......+.+..+.......+...+...............+..+.+...+......+...+........+....+......+............+..+...+.......+...........+...+.+....................+......+.......+..+......+.......+..+....+..+....+........+...+...+....+......+...........+.............+.....................+..+...+.........+.+...........+...+.+......+......+..+..........+......+......+.....+...+..........+...+.....+...+....+..+......+...+.......+............+.....+......+..........+......+.....+................+..+......+.+......+........+.......+...+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
..+..+.+.....+.........+...+.+...............+.....+....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*..+.....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.....+...+...+...+....+......+......+.....+.+...+......+........+..................+...+.+...+..................+.................+...+.......+...+............+..+...+......+.+.....+.+...+.................+....+.........+.........+..+....+..............+...+...+....+...+.........+...+...+..+.......+.....+.+.....+....+..............+......+....+.........+...+.........+.....+...+.+.........+............+....................+..........+..............+.......+...+..+...+....+..+.+.....+.......+..+....+.....+.+...........+...+....+..+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-----
```